valohai_cli/packager.py

```diff
@@ -148,14 +155,24 @@ def _get_files_walk(dir: str) -> Iterable[Tuple[str, str]]:
             yield (file_rel_path, file_abs_path)
 
 
-def _get_files(dir: str, allow_git: bool = True) -> Tuple[bool, Iterable[Tuple[str, str]]]:
-    if allow_git and os.path.exists(os.path.join(dir, '.git')):
-        # We have .git, so we can try to use Git to figure out a file list of nonignored files
-        try:
-            return (True, _get_files_with_git(dir))  # return the generator
-        except subprocess.CalledProcessError as cpe:
-            warn(f'.git exists, but we could not use git ls-files (error {cpe.returncode}), falling back to non-git')
-    return (False, _get_files_walk(dir))  # return the generator
+def _get_files(dir: str, allow_git: bool = True) -> Tuple[GitUsage, Iterable[Tuple[str, str]]]:
+    gitignore_path = os.path.join(dir, '.gitignore')
+
+    if allow_git:
+        if os.path.exists(os.path.join(dir, '.git')):
+            # We have .git, so we can try to use Git to figure out a file list of nonignored files
+            try:
+                return (GitUsage.GIT_LS_FILES, _get_files_with_git(dir))  # return the generator
+            except subprocess.CalledProcessError as cpe:
+                warn(f'.git exists, but we could not use git ls-files (error {cpe.returncode}), falling back to non-git')
+
+        # Limited support of .gitignore even without git
+        elif os.path.exists(gitignore_path):
+            with open(gitignore_path, "r") as git_ignore_file:
+                ignore_patterns = [pattern.strip() for pattern in git_ignore_file.readlines()]
+                return (GitUsage.GITIGNORE_WITHOUT_GIT, (p for p in _get_files_walk(dir) if is_valid_path(p[0], ignore_patterns)))
```

Comment:
No need to keep the file open during this time.
```suggestion
            return (GitUsage.GITIGNORE_WITHOUT_GIT, (p for p in _get_files_walk(dir) if is_valid_path(p[0], ignore_patterns)))
```